Traceback (most recent call last):
  File "HAPLOID.py", line 317, in <module>
    displayResults(results)
  File "HAPLOID.py", line 303, in displayResults
    qvalues = multipletests(pvalues, is_sorted=True, returnsorted=True)
TypeError: multipletests() got an unexpected keyword argument 'is_sorted'
Traceback (most recent call last):
  File "HAPLOID.py", line 318, in <module>
    displayResults(results)
  File "HAPLOID.py", line 305, in displayResults
    result.append(str(qval))
UnboundLocalError: local variable 'result' referenced before assignment
Traceback (most recent call last):
  File "HAPLOID.py", line 319, in <module>
    displayResults(results)
  File "HAPLOID.py", line 306, in displayResults
    result.append(str(qval))
UnboundLocalError: local variable 'result' referenced before assignment
Traceback (most recent call last):
  File "HAPLOID.py", line 319, in <module>
    displayResults(results)
  File "HAPLOID.py", line 305, in displayResults
    for qval in qvalues[1][0]:
TypeError: 'numpy.float64' object is not iterable
Traceback (most recent call last):
  File "HAPLOID.py", line 319, in <module>
    displayResults(results)
  File "HAPLOID.py", line 305, in displayResults
    for qval in qvalues[1][0]:
TypeError: 'numpy.float64' object is not iterable
Traceback (most recent call last):
  File "HAPLOID.py", line 320, in <module>
    displayResults(results)
  File "HAPLOID.py", line 307, in displayResults
    result.append(str(qval))
UnboundLocalError: local variable 'result' referenced before assignment
Traceback (most recent call last):
  File "HAPLOID.py", line 318, in <module>
    displayResults(results)
  File "HAPLOID.py", line 307, in displayResults
    print '<tr>{}</tr>'.format('<td>'+'</td><td>'.join(result)+'</td>')
TypeError: sequence item 11: expected string, numpy.ndarray found
Traceback (most recent call last):
  File "HAPLOID.py", line 317, in <module>
    displayResults(results)
  File "HAPLOID.py", line 303, in displayResults
    qvalues = multipletests(pvalues, alpha=0.05, method='holm-sidak')
  File "/home4/dmacmill/anaconda/lib/python2.7/site-packages/statsmodels/stats/multitest.py", line 132, in multipletests
    alphacSidak = 1 - np.power((1. - alphaf), 1./ntests)
ZeroDivisionError: float division by zero
Traceback (most recent call last):
  File "HAPLOID.py", line 319, in <module>
    displayResults(results)
  File "HAPLOID.py", line 308, in displayResults
    print '<tr>{}</tr>'.format('<td>'+'</td><td>'.join(result)+'</td>')
TypeError: sequence item 11: expected string, numpy.ndarray found
Traceback (most recent call last):
  File "HAPLOID.py", line 21, in <module>
    hlacountfilter = int(form.getvalue("hlacountfilter"))
TypeError: int() argument must be a string or a number, not 'NoneType'
Traceback (most recent call last):
  File "HAPLOID.py", line 273, in <module>
    results = analyzeHLAs(patients, uniquehlas, hlacountfilter)
  File "HAPLOID.py", line 191, in analyzeHLAs
    print results[uhla][pos]['tt'] + results[uhla][pos]['ft']
TypeError: unsupported operand type(s) for +: 'dict' and 'dict'
Traceback (most recent call last):
  File "HAPLOID.py", line 21, in <module>
    hlacountfilter = int(form.getvalue("hlacountfilter"))
TypeError: int() argument must be a string or a number, not 'NoneType'
Traceback (most recent call last):
  File "HAPLOID.py", line 273, in <module>
    results = analyzeHLAs(patients, uniquehlas, hlacountfilter)
  File "HAPLOID.py", line 191, in analyzeHLAs
    print results[uhla][pos]['tt'] + results[uhla][pos]['ft']
TypeError: unsupported operand type(s) for +: 'dict' and 'dict'
Traceback (most recent call last):
  File "HAPLOID.py", line 274, in <module>
    results = analyzeHLAs(patients, uniquehlas, hlacountfilter)
  File "HAPLOID.py", line 191, in analyzeHLAs
    for aa in results[uhla][pos]['tt'] + results[uhla][pos]['ft']:
TypeError: unsupported operand type(s) for +: 'dict' and 'dict'
Traceback (most recent call last):
  File "HAPLOID.py", line 274, in <module>
    results = analyzeHLAs(patients, uniquehlas, hlacountfilter)
  File "HAPLOID.py", line 191, in analyzeHLAs
    for aa in results[uhla][pos]['tt'] + results[uhla][pos]['ft']:
TypeError: unsupported operand type(s) for +: 'dict' and 'dict'
Traceback (most recent call last):
  File "HAPLOID.py", line 274, in <module>
    results = analyzeHLAs(patients, uniquehlas, hlacountfilter)
  File "HAPLOID.py", line 191, in analyzeHLAs
    for aa in results[uhla][pos]['tt'] + results[uhla][pos]['ft']:
TypeError: unsupported operand type(s) for +: 'dict' and 'dict'
Traceback (most recent call last):
  File "HAPLOID.py", line 283, in <module>
    results = analyzeHLAs(patients, uniquehlas, hlacountfilter)
  File "HAPLOID.py", line 164, in analyzeHLAs
    for pos in xrange(patients[patient]['seq']):
TypeError: an integer is required
Traceback (most recent call last):
  File "HAPLOID.py", line 283, in <module>
    results = analyzeHLAs(patients, uniquehlas, hlacountfilter)
  File "HAPLOID.py", line 168, in analyzeHLAs
    if aa in seqcounts[pos]:
TypeError: argument of type 'int' is not iterable
Traceback (most recent call last):
  File "HAPLOID.py", line 282, in <module>
    results = analyzeHLAs(patients, uniquehlas, hlacountfilter)
  File "HAPLOID.py", line 167, in analyzeHLAs
    if aa in seqcounts[pos]:
TypeError: argument of type 'int' is not iterable
Traceback (most recent call last):
  File "HAPLOID.py", line 283, in <module>
    results = analyzeHLAs(patients, uniquehlas, hlacountfilter)
  File "HAPLOID.py", line 168, in analyzeHLAs
    if aa in seqcounts[pos]:
TypeError: argument of type 'int' is not iterable
Traceback (most recent call last):
  File "HAPLOID.py", line 282, in <module>
    results = analyzeHLAs(patients, uniquehlas, hlacountfilter)
  File "HAPLOID.py", line 181, in analyzeHLAs
    if [x for x in patients[patient]['seq'][pos]].count(aa) < minAA:
NameError: global name 'minAA' is not defined
Traceback (most recent call last):
  File "HAPLOID.py", line 281, in <module>
    results = analyzeHLAs(patients, uniquehlas, hlacountfilter, aacountfilter)
  File "HAPLOID.py", line 180, in analyzeHLAs
    if seqcount[pos][aa] < aacountfilter:
NameError: global name 'seqcount' is not defined
Traceback (most recent call last):
  File "HAPLOID.py", line 283, in <module>
    displayResults(results)
  File "HAPLOID.py", line 267, in displayResults
    qvalues = multipletests(pvalues, alpha=0.05, method='holm-sidak')
  File "/home4/dmacmill/anaconda/lib/python2.7/site-packages/statsmodels/stats/multitest.py", line 132, in multipletests
    alphacSidak = 1 - np.power((1. - alphaf), 1./ntests)
ZeroDivisionError: float division by zero
Traceback (most recent call last):
  File "HAPLOID.py", line 282, in <module>
    displayResults(results)
  File "HAPLOID.py", line 266, in displayResults
    qvalues = multipletests(pvalues, alpha=0.05, method='holm-sidak')
  File "/home4/dmacmill/anaconda/lib/python2.7/site-packages/statsmodels/stats/multitest.py", line 132, in multipletests
    alphacSidak = 1 - np.power((1. - alphaf), 1./ntests)
ZeroDivisionError: float division by zero
Traceback (most recent call last):
  File "HAPLOID.py", line 282, in <module>
    displayResults(results)
  File "HAPLOID.py", line 266, in displayResults
    qvalues = multipletests(pvalues, alpha=0.05, method='holm-sidak')
  File "/home4/dmacmill/anaconda/lib/python2.7/site-packages/statsmodels/stats/multitest.py", line 132, in multipletests
    alphacSidak = 1 - np.power((1. - alphaf), 1./ntests)
ZeroDivisionError: float division by zero
Traceback (most recent call last):
  File "HAPLOID.py", line 23, in <module>
    maxp = int(form.getvalue("maxp"))
ValueError: invalid literal for int() with base 10: '0.05'
Traceback (most recent call last):
  File "HAPLOID.py", line 23, in <module>
    maxp = int(form.getvalue("maxp"))
ValueError: invalid literal for int() with base 10: '0.05'
Traceback (most recent call last):
  File "HAPLOID.py", line 290, in <module>
    displayResults(results,maxq)
  File "HAPLOID.py", line 272, in displayResults
    qvalues = multipletests(pvalues, alpha=0.05, method='holm-sidak')
  File "/home4/dmacmill/anaconda/lib/python2.7/site-packages/statsmodels/stats/multitest.py", line 132, in multipletests
    alphacSidak = 1 - np.power((1. - alphaf), 1./ntests)
ZeroDivisionError: float division by zero
Traceback (most recent call last):
  File "HAPLOID.py", line 293, in <module>
    displayResults(results,maxq)
  File "HAPLOID.py", line 277, in displayResults
    for i in xrange(len(qvalues[1])):
UnboundLocalError: local variable 'qvalues' referenced before assignment
Traceback (most recent call last):
  File "HAPLOID.py", line 293, in <module>
    displayResults(results,maxq)
  File "HAPLOID.py", line 277, in displayResults
    for i in xrange(len(qvalues[1])):
UnboundLocalError: local variable 'qvalues' referenced before assignment
Traceback (most recent call last):
  File "HAPLOID.py", line 285, in <module>
    displayResults(results,maxq,maxp)
  File "HAPLOID.py", line 269, in displayResults
    for i in xrange(len(qvalues[1])):
NameError: global name 'qvalues' is not defined
Traceback (most recent call last):
  File "HAPLOID.py", line 23, in <module>
    maxp = float(form.getvalue("maxp"))
TypeError: float() argument must be a string or a number
Traceback (most recent call last):
  File "HAPLOID.py", line 23, in <module>
    maxp = float(form.getvalue("maxp"))
TypeError: float() argument must be a string or a number
Traceback (most recent call last):
  File "HAPLOID.py", line 313, in <module>
    displayResults(results,maxq,maxp)
  File "HAPLOID.py", line 289, in displayResults
    qvalues = multipletests(pvalues, alpha=0.05, method='fdr_bh')
  File "/home4/dmacmill/anaconda/lib/python2.7/site-packages/statsmodels/stats/multitest.py", line 132, in multipletests
    alphacSidak = 1 - np.power((1. - alphaf), 1./ntests)
ZeroDivisionError: float division by zero
Traceback (most recent call last):
  File "HAPLOID.py", line 311, in <module>
    results = sorted(results, key = lambda x: (int(x[1]), x[0], -x[3], x[2]))
  File "HAPLOID.py", line 311, in <lambda>
    results = sorted(results, key = lambda x: (int(x[1]), x[0], -x[3], x[2]))
TypeError: bad operand type for unary -: 'str'
